src/js/gi-bill-comparison-tool/components/Dropdown.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+class Dropdown extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: props.defaultOption
+    };
+  }
+
+  render() {
+    if (!this.props.visible) {
+      return null;
+    }
+    return (
+      <div id={this.props.identifier + '-form'} className={this.props.className}>
+        {this.renderLabel()}
+        <select id={this.props.identifier}
+            name={this.props.identifier.replace(/\-/g, '_')}
+            alt={this.props.alt}
+            onChange={(e) => { this.props.onChange(e) && this.setState({ value: event.target.value }); }}
+            defaultValue={this.props.defaultOption}>
+          {this.renderDropdownOptions(this.props.options)}
+        </select>
+      </div>
+    );
+  }
+
+  renderLabel() {
+    if (!this.props.showLabel) {
+      return null;
+    }
+    return this.props.children;
+  }
+
+  renderDropdownOptions(optionsArray) {
```

Comment:
Although it does increase the complexity of this one clear-purpose function, it really simplifies the code where these Dropdowns are used.